eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1622,6 +1622,17 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       d.commitments.resolveCommitment(tx) match {
         case Some(commitment) =>
           log.warning("a commit tx for fundingTxIndex={} fundingTxId={} has been confirmed", commitment.fundingTxIndex, commitment.fundingTxId)
+          // Funding transactions with a greater index will never confirm: we abandon them to unlock their wallet inputs,
+          // which would otherwise stay locked forever in our bitcoind wallet.
+          d.commitments.all
+            .collect { case c: Commitment if commitment.fundingTxIndex <= c.fundingTxIndex => c.fundingTxId }
+            .foreach { txId => wallet.abandon(txId) }
+          // Any anchor transaction that we created based on the latest local or remote commit will never confirm either
+          // so we need to abandon them to unlock their wallet inputs.
+          nodeParams.db.audit.listPublished(d.channelId).collect {
```

Comment:
It bothers me a little bit that we are calling the audit db here, because the audit db is meant to live on the side, not be actively involved in the normal operations. I like the fact that we are not creating an additional table though.

Reply:
I agree, that's part of my hesitation: do we create another `channel_transactions` table and use it here? I figured that for now since we had everything already in the audit table, it probably wasn't worth it and we could defer this to later when we have more scenarios that need it, but I'm not 100% decided what's best.

Reply:
Fair enough, let's start with this then.

Reply:
I think that when we move to 0-fee commitment transactions, then we will likely need to introduce a DB table for our force-close transactions (because since we will get rid of the CSV 1 in commit outputs, we may need to abandon more transaction types during conflicts). That will probably be the right time to clean that up and use this new table instead of the audit one.